packages/core/src/ceramic.ts

```diff
@@ -230,10 +241,10 @@ class Ceramic implements CeramicApi {
 
   /**
    * Create Ceramic instance
-   * @param ipfs - IPFS instance
+   * @param components - Ceramic internal components provided to `create` from higher level
    * @param config - Ceramic configuration
    */
-  static async create(ipfs: IpfsApi, config: CeramicConfig = {}): Promise<Ceramic> {
+  static async create(components: CeramicProvidedComponents, config: CeramicConfig = {}): Promise<Ceramic> {
```

Comment:
I guess technically this is a breaking API change. Is the main reason we're breaking them apart that it will be easier to log the config? Wouldn't it be better to just filter the config object in that case?

Reply:
Yeah, there's no technical reason why this *has* to be broken out.  We certainly could continue to just add more things into the `CeramicConfig`.  It's less about the filtering in my mind and more about logically separating where we're providing configuration options to Ceramic, vs where we plug in implementations of entire subsystems.  I already need to add a way to provide a logger instance (while continuing to create the logger internally if no logger instance is provided, but config options about how to construct one are), and I'm worried that it'll be confusing to have a full logger instance and the config options that describe how to *construct* a logger instance in the same object.  We also have other places where we allow providing an entire instance of a subcomponent rather than just config options that control what subcomponents get used, with more to come.  The different `StateStore` backends are a good example.  Is an entire implementation of the `StateStore` interface really a config option?  Or should we have a separate mechanism for pluggable components that extend the core functionality?

I could go either way on this one.  Would be curious if @ukstv has an opinion here

Reply:
Another option could be to have two properties in the config type. One for modules, and one for simple configs?
Idk what's best here tbh, but it feels a bit weird to have two separate types.

Reply:
> Another option could be to have two properties in the config type.

You mean two properties at the top-level, with everything else moving into sub-objects?  So basically changing `CeramicConfig` to look something like:

```
export interface CeramicConfig {
  flags: {
    ethereumRpcUrl?: string;
    anchorServiceUrl?: string;
    pinsetDirectory?: string;
    // ... <the rest of the current CeramicConfig properties>
  };
  components: {
    ipfs: IpfsApi;
    logger?: DiagnosticLogger;
    stateStore?: StateStore;
    // ...
  };
}
```

Reply:
Yeah something like that. Maybe call them `modules` and `params`?